### PR TITLE
[Fix] Drop the duplicated DSpark draft sample_block call

### PR DESCRIPTION
**Pull request by hnyls2002**
`DsparkDraftSampler.__call__` runs `sample_block` twice: the call added inside the `if draft_tokens is None:` block (#33561) plus the original trailing one, which overwrites it. `self.out` came from the second call and `self.corrected_out` from the first, so the emitted draft tokens and the corrected logits were drawn from two separate passes.

Note this activates the fused greedy fast path, whose result was previously computed and then discarded.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #33223878538](https://github.com/sgl-project/sglang/actions/runs/33223878538)
Latest PR Test (Extra): :x: [Run #33223878487](https://github.com/sgl-project/sglang/actions/runs/33223878487)
Latest PR Test (AMD ROCm 7.2): :no_entry_sign: [Run #33223878558](https://github.com/sgl-project/sglang/actions/runs/33223878558)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test registered/core/test_basic_sanity_dspark.py registered/spec/dspark/test_dspark_kernel_parity.py registered/spec/dspark/test_dspark_stacked_ctx_kv_parity.py
